manubot/cite/util.py

```diff
@@ -218,101 +218,106 @@ def is_valid_citation(
+                f'Source {source!r} is not valid.\n'
                 f'Valid citation sources are {{{", ".join(sorted(sources))}}}'
             )
         return False
 
-    inspection = inspect_citation_identifier(citation)
+    inspection = inspect_citekey(citekey)
     if inspection:
-        logging.error(f'invalid {source} citation: {citation}\n{inspection}')
+        logging.error(f'invalid {source} citekey: {citekey}\n{inspection}')
         return False
 
     return True
 
 
-def get_citation_short_id(standard_id):
+def shorten_citekey(citekey):
     """
-    Get the short_id derived from a citation's standard_id.
-    Short IDs are generated by converting the standard_id to a 6 byte hash,
-    and then converting this digest to a base62 ASCII str. The output
-    short_id consists of characters in the following ranges: 0-9, a-z and A-Z.
+    Return a shortened citekey derived from the input citekey.
+    The input citekey should be standardized prior to this function,
+    since differences in the input citekey will result in different shortened citekeys.
+    Short citekeys are generated by converting the input citekey to a 6 byte hash,
+    and then converting this digest to a base62 ASCII str. Shortened
+    citekeys consist of characters in the following ranges: 0-9, a-z and A-Z.
     """
     import hashlib
     import base62
-    assert '@' not in standard_id
-    as_bytes = standard_id.encode()
+    assert '@' not in citekey
+    as_bytes = citekey.encode()
     blake_hash = hashlib.blake2b(as_bytes, digest_size=6)
     digest = blake_hash.digest()
-    short_id = base62.encodebytes(digest)
-    return short_id
+    short_citekey = base62.encodebytes(digest)
+    return short_citekey
 
 
-def citation_to_citeproc(citation, prune=True):
+def citekey_to_csl_item(citekey, prune=True):
     """
-    Return a dictionary with citation metadata
+    Generate a CSL Item (Python dictionary) for the input citekey.
     """
-    citation == standardize_citation(citation, warn_if_changed=True)
-    source, identifier = citation.split(':', 1)
+    citekey == standardize_citekey(citekey, warn_if_changed=True)
+    source, identifier = citekey.split(':', 1)
 
     if source in citeproc_retrievers:
         citeproc_retriever = import_function(citeproc_retrievers[source])
         csl_item = citeproc_retriever(identifier)
     else:
-        msg = f'Unsupported citation source {source} in {citation}'
+        msg = f'Unsupported citation source {source!r} in {citekey!r}'
         raise ValueError(msg)
 
     from manubot import __version__ as manubot_version
     from manubot.cite.citeproc import (
-        citeproc_passthrough,
+        csl_item_passthrough,
         append_to_csl_item_note,
     )
 
     note_text = f'This CSL JSON Item was automatically generated by Manubot v{manubot_version} using citation-by-identifier.'
     note_dict = {
-        'standard_id': citation,
+        'standard_id': citekey,
```

Comment:
Are we intentionally keeping the `standard_id` terminology here for now?  It appears to be intended that way per the `csl_item_set_standard_id` docs.

Reply:
That is the current implementation. It has two primary motivations:

1. since CSL using the term `id` for the citekey, it fits the CSL JSON vernacular.
2. easiest for backwards compatibility since we may already have manual-references that are using the `standard_id` in notes.

We could change it at a later or now if you think it's preferable to use `standard_citekey`, `original_standard_citekey`, and `original_citekey` in the notes.

Reply:
Because CSL uses the `id` term and the rest of Manubot now uses "citekeys", we'll have to convert between these terms somewhere.  The docstring does explain the correspondence
> The standard_id is extracted from a "standard_citation" field, the "note" field, or the "id" field.

> Note that the Manubot software generally refers to the "id" of a CSL Item as a citekey.
 However, in this context, we use "id" rather than "citekey" for consistency with CSL's "id" field.

I'm undecided about what is least disruptive.  Currently, there are two sets of terms describing the progression from original ids/citekeys to standardized ids/citekeys in `get_citekeys_df` and `csl_item_set_standard_id`.  When reading the CSL JSON, we can have `original_standard_id` or `original_id`.  When reading the manuscript, we have `manuscript_citekey`.  Should these be harmonized?

I also forgot when Manubot would encounter CSL JSON with the `standard_citation` field set.  Was that written by older versions of Manubot?  Or part of the original process for manual references?

Reply:
> When reading the CSL JSON, we can have original_standard_id or original_id. When reading the manuscript, we have manuscript_citekey. Should these be harmonized?

`original_id` and `original_standard_id` refer to the citekey workflow of a reference whereas `manuscript_citekey` refers to the citekey workflow of a citation. In https://github.com/manubot/manubot/issues/114, I'd like to come to a more harmonized nomenclature and processing workflow. But in the meantime, I think we should proceed with this PR without changing `original_standard_id` and `original_id`. I am learning towards changing them, but not right now and not without a little more thought.

> I also forgot when Manubot would encounter CSL JSON with the standard_citation field set. Was that written by older versions of Manubot? Or part of the original process for manual references?

Previously, manual references would set `standard_citation`, but now the recommendation is to use `id`.